### PR TITLE
Refactor error handling in sh_histinit function

### PR DESCRIPTION
**Pull request by siteshwar**
Resolves: cid#294200

### CODE REVIEW

File: src/cmd/ksh93/edit/history.c [hunk 1]
```diff
@@ -218,13 +218,14 @@ int sh_histinit(void *sh_context) {
         if (shgd->userid)
 #endif  // KSHELL
         {
-            if (!(fname = pathtmp(NULL, 0, 0, NULL))) return 0;
+            if (!(fname = pathtmp(NULL, 0, 0, NULL))) goto fail;
```
Comment (krader1961) [inline src/cmd/ksh93/edit/history.c]: There is no good reason to introduce a bunch or `goto`'s. That label does nothing but return. We want to remove goto's not add them; at least not without good reason. Other than that adding `(void)` to silence the unused `fcntl()` retval error is fine. Feel free to merge that single line change without the goto's.

Comment (siteshwar) [reply src/cmd/ksh93/edit/history.c]: Isn't it better to do error handling at just one place ?

Comment (siteshwar) [reply src/cmd/ksh93/edit/history.c]: Anyways, I will open a separate pull request to fix all these warnings.

Comment (krader1961) [reply src/cmd/ksh93/edit/history.c]: > Isn't it better to do error handling at just one place ?

If there was some actual error handling, sure. But in this case there is none. Long ago I worked with some programmers who believed a function should have at most to `return` statements: one for success and one for failure. That resulted in some awful code. There is nothing wrong with multiple return points within a function.

**Pull request closed without merging**